docs/walkthrough.md

```diff
@@ -0,0 +1,210 @@
+
+## Create, Update or Delete data for an Entity
+
+Using our example _Customer schema_ the following requests show how we can create, update or delete data for a specific entity calling the existing API endpoints.
+
+### Create
+
+```json
+> POST /api/customer/save
+> Host: localhost:8000
+> Authorization: Token x
+
+{
+ 	"version": "v1",
+ 	"key": "c60eba64",
+ 	"payload": {
+ 		"firstName": "John",
+ 		"lastName": "Doe"
+ 	}
+}
+```
+
+```json
+{
+  "message": "entity with key c60eba64 created in version v1"
+}
+```
+
+### Update
+
+To update an entity you don't need to send the whole data again, but just the fields you need to change.
+
+```json
+> PATCH /api/customer/c60eba64/v1/update
+> Host: localhost:8000
+> Authorization: Token x
+
+{
+ 	"lastName": "Doe"
+}
+```
+
+```json
+{
+  "message": "entity with key c60eba64 updated in version v1"
+}
+```
+
+### Delete
+
+We can either delete just a specific version of an entity or if we _omit_ the version in the request payload, all the versions for the entity key provided will be deleted.
+
+```json
+> POST /api/customer/delete
+> Host: localhost:8000
+> Authorization: Token x
+
+{
+ 	"version": "v1",
+ 	"key": "c60eba64"
+}
+```
+
+```json
+{
+  "message": "entity with key c60eba64 deleted from version v1"
+}
+```
+
+or
+
+```json
+> POST /api/customer/delete
+> Host: localhost:8000
+> Authorization: Token x
+
+{
+ 	"key": "c60eba64"
+}
+```
+
+```json
+{
+  "message": "entity with key c60eba64 deleted from all versions"
+}
+```
+
+## Entity data Validation
+
+Every time you call an endpoint sending data to a specific schema this data is validated against the _JSON schema_ you initially loaded into Herbie.
+
+Using Herbie you guarantee that you can have a central definition of what an entity looks like and validate the data against the schema every time there is a change to an entity on your system.
+
+```json
+> POST /api/customer/save
+> Host: localhost:8000
+> Authorization: Token x
+
+{
+ 	"version": "v1",
+ 	"key": "c60eba64",
+ 	"payload": {
+ 		"firstName": "John"
+ 	}
+}
+```
+
+```json
+{
+  "message": {
+    "lastName": {
+      "error_message": "'lastName' is a required property",
+      "validation_error": "required"
+    }
+  }
+}
+```
+
+## Schema Changes and Backward Compatibility
+
+Of course our _customer_v1_ is a really simple and dummy definition of what a customer entity could/should look like.
+
+Having in mind that these definitions change along the way Herbie allows to either change an existing schema version or create a new version for the same entity.
+
+In case you are replacing the same version we will check if the new schema is _backwards compatible_, meaning if the old data is still valid for this new schema.
+
+If we, for example, change our _customer_v1_ and change the _firstName_ to be a required field this would introduce a BC break as previously that field was not required.
```

Comment:
Maybe highlight here also that we don't allow BC breaks and through an exception in that case.